### PR TITLE
Disable the file field block by setting max number restriction

### LINKED ISSUE #2102: Merge the two file upload field type in to one.
There are now two stream fields for uploading files, "File field" and "Multiple file field".

We should remove/hide "File field" from being added to new forms. Instead add a setting on "Multiple file field" to set how many fields are allowed. 1-10 or unlimited are my suggestion for options.

Or is it never needed to limit the number of uploads? If that is the case the new setting is not needed.

Lets use the icon now used for "File field" on "Multiple file field" as well when this is worked on.

### PR DESCRIPTION
**Pull request by sandeepsajan0**
Fixes #2102 

With the custom max_no restrictions, it will look like 

image

### CODE REVIEW

**Comment by sks444** [comment]
This looks good but it only works when file field block is select at least once even for creating new application forms.

**Comment by sandeepsajan0** [comment]
@sks444 @frjo I think I found a fix for caching issue as well. It caches the form inside edit handler but it still calls the Create class get_form method. So, we can change the form's meta (max number for files) there. Let me test it and push it.

**Comment by sandeepsajan0** [comment]
@sks444 Can you please test it again, it seems working fine for me now.

**Comment by frjo** [comment]
Deploying to test site now.

File: hypha/apply/funds/blocks.py [hunk 1]
```diff
@@ -194,6 +194,9 @@ class ApplicationCustomFormFieldsBlock(CustomFormFieldsBlock, FormFieldsBlock):
     required_blocks = ApplicationMustIncludeFieldBlock.__subclasses__()
     single_blocks = ApplicationSingleIncludeFieldBlock.__subclasses__()
 
+    class Meta:
+        block_counts = {}
+
```
Comment (sks444) [inline hypha/apply/funds/blocks.py]: Is this still required?

File: hypha/apply/funds/admin_views.py [hunk 2]
```diff
@@ -101,13 +101,24 @@ def get_initial(self):
...
+        return form
+
     def form_invalid(self, form):
         show_admin_form_error_messages(self.request, form)
         return self.render_to_response(self.get_context_data(form=form))
 
 
 class EditApplicationFormView(EditView):
 
+    def get_form(self):
```
Comment (sks444) [inline hypha/apply/funds/admin_views.py:120]: Add a comment on why this change is required and what it does.

File: hypha/apply/funds/admin_views.py [hunk 3]
```diff
@@ -101,13 +101,24 @@ def get_initial(self):
 
 class CreateApplicationFormView(CreateView):
 
+    def get_form(self):
```
Comment (sks444) [inline hypha/apply/funds/admin_views.py:104]: Add a comment on why this change is required and what it does.

**Review by sks444 — APPROVED** [review]
Looks good. Nice work @sandeepsajan0!

File: hypha/apply/funds/blocks.py (hunk 1 above)
Comment (sandeepsajan0) [reply hypha/apply/funds/blocks.py]: No, we don't need it. I left it to avoid error but I just tested again by removing it and everything worked fine without this. I'll remove it and will add comments about the change as well.
